src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixPlugin.groovy

```diff
@@ -90,31 +87,43 @@ class ScalafixPlugin implements Plugin<Project> {
                 prop.split('\\s*,\\s*').findAll { !it.isEmpty() }.toList()
             }))
             scalafixTask.mode = taskMode
-            scalafixTask.scalaVersion = sourceSet.scalaVersion.get()
-            scalafixTask.classpath = sourceSet.fullClasspath.collect { it.path }
-            scalafixTask.compileOptions = sourceSet.compilerOptions
-            scalafixTask.semanticdbConfigured = extension.isSemanticdbEnabled()
+            scalafixTask.sourceSetName = sourceSet.name
+            scalafixTask.scalaVersion.set(project.provider({ resolveScalaVersion(sourceSet, extension) }))
+            scalafixTask.classpath.set(project.provider( { sourceSet.fullClasspath.collect { it.path } }))
+            scalafixTask.compileOptions.set(project.provider({ sourceSet.compilerOptions }))
+            scalafixTask.semanticdbConfigured = extension.semanticdbEnabled
 
-            if (extension.isSemanticdbEnabled()) {
+            if (extension.semanticdbEnabled) {
                 scalafixTask.dependsOn sourceSet.compileTask
             }
         })
 
         mainTask.dependsOn taskProvider
     }
 
-    private void configureSemanticdbCompilerPlugin(Project project, ScalaSourceSet sourceSet) {
-        def semanticDbCoordinates = ScalafixProps.getSemanticDbArtifactCoordinates(sourceSet.scalaVersion.get(),
-                Optional.ofNullable(project.scalafix.semanticdb.version.getOrNull()))
+    private void configureSemanticdbCompilerPlugin(Project project, ScalaSourceSet sourceSet, ScalafixExtension extension) {
+        def scalaVersion = resolveScalaVersion(sourceSet, extension)
+
+        if (scalaVersion == null || scalaVersion.blank) {
```

Comment:
looks like some tests for Gradle 4.x failed because `.blank` doesn't exist in older versions of Groovy 😞 . I will need to change this

Reply:
this might help and is a stricter match `!(scalaVersion ==~ "\d+\.\d+\.\d+"`)`
or "scalaVersion ==~ "\s*"" for the isblank